docs/dev/design/new-notifications-system.rst

```diff
@@ -0,0 +1,459 @@
+            type = str(ERROR, WARINIG, NOTE, TIP)
+
+        def get_display_icon(self):
+            if self.icon:
+                return self.icon
+
+            if self.type == ERROR:
+                return "fa-exclamation"
+            if self.type == WARNING:
+                return "fa-triangle-exclamation"
+
+        def get_display_color(self):
+            if self.color:
+                return color
+
+            if self.type == ERROR:
+                return "red"
+            if self.type == WARNING:
+                return "yellow"
+
+
+
+Definition of notifications to display to users
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This constant defines all the possible notifications to be displayed to the user.
+Each notification has to be defined here using the ``Message`` class previously defined.
+
+.. code-block:: python
+
+    NOTIFICATION_MESSAGES = {
+        "generic-with-build-id": Message(
+            header=_("Unknown problem"),
+            # Note the message receives the instance it's attached to
+            # and could be use it to inject related data
+            body=_(
+                """
+          There was a problem with Read the Docs while building your documentation.
+          Please try again later.
+          If this problem persists,
+          report this error to us with your build id ({instance[pk]}).
+        """,
+                type=ERROR,
+            ),
+        ),
+        "build-os-required": Message(
+            header=_("Invalid configuration"),
+            body=_(
+                """
+          The configuration key "build.os" is required to build your documentation.
+          <a href='https://docs.readthedocs.io/en/stable/config-file/v2.html#build-os'>Read more.</a>
+        """,
+                type=ERROR,
+            ),
+        ),
+        "cancelled-by-user": Message(
+            header=_("User action"),
+            body=_(
+                """
+          Build cancelled by the user.
+        """,
+                type=ERROR,
+            ),
+        ),
+        "os-ubuntu-18.04-deprecated": Message(
+            header=_("Deprecated OS selected"),
+            body=_(
+                """
+          Ubuntu 18.04 is deprecated and will be removed soon.
+          Update your <code>.readthedocs.yaml</code> to use a newer image.
+        """,
+                type=TIP,
+            ),
+        ),
+    }
+
+
+
+``Notification`` model definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This class is the representation of a notification attached to an resource (e.g. User, Build, etc) in the database.
+It contains an identifier (``slug``) pointing to one of the messages defined in the previous section (key in constant ``NOTIFICATION_MESSAGES``).
+
+.. code-block:: python
+
+    import textwrap
+    from django.utils.translation import gettext_noop as _
+
+
+    class Notification(TimeStampedModel):
+        # Message identifier
+        slug = models.CharField(max_length=128)
```

Comment:
This field could be more descriptive, `slug` is overused in our code and is used more commonly as an alternative primary key. It doesn't seem like you are describing a primary key here, but instead referencing the notification message id. So, for that, I think `Notification.message_id` would make the most sense.

Reply:
Done in 26960553